### ISSUE TITLE
**Facing dimension problem, when I include  Dropout into the network**

### ISSUE THREAD

**Issue by badripatro** [torch/nn#1300]
Problem: **Facing dimension problem, when I include  Dropout into the network**. With out droout ther is no dimension issue
--------------------------------------------------------------------------------------------------------
**Error** 
-----------------------------------------------------------------------------
/users/gpu/badri1/torch/install/bin/luajit: .../gpu/badri1/torch/install/share/lua/5.1/nn/Container.lua:67: 
In 7 module of nn.Sequential:
/users/gpu/badri1/torch/install/share/lua/5.1/nn/THNN.lua:110: **Need gradOutput of dimension 4 and gradOutput.size[1] == 128 but got gradOutput to be of shape: [10 x 64 x 112 x 112] at /tmp/luarocks_cunn-scm-1-7572/cunn/lib/THCUNN/generic/SpatialAveragePooling.cu:66**
stack traceback:
	[C]: in function 'v'
	/users/gpu/badri1/torch/install/share/lua/5.1/nn/THNN.lua:110: in function 'SpatialAveragePooling_updateGradInput'
	...torch/install/share/lua/5.1/nn/SpatialAveragePooling.lua:67: in function 'updateGradInput'
	/users/gpu/badri1/torch/install/share/lua/5.1/nn/Module.lua:31: in function </users/gpu/badri1/torch/install/share/lua/5.1/nn/Module.lua:29>
	[C]: in function 'xpcall'
	.../gpu/badri1/torch/install/share/lua/5.1/nn/Container.lua:63: in function 'rethrowErrors'
	...gpu/badri1/torch/install/share/lua/5.1/nn/Sequential.lua:84: in function 'backward'
	train.lua:763: in function 'lossFun'
	train.lua:868: in function 'Train'
	train.lua:999: in main chunk
	[C]: in function 'dofile'
	...dri1/torch/install/lib/luarocks/rocks/trepl/scm-1/bin/th:150: in main chunk
	[C]: at 0x00406670

WARNING: If you see a stack trace below, it doesn't point to the place where this error occurred. Please use only the one above.
stack traceback:
	[C]: in function 'error'
	.../gpu/badri1/torch/install/share/lua/5.1/nn/Container.lua:67: in function 'rethrowErrors'
	...gpu/badri1/torch/install/share/lua/5.1/nn/Sequential.lua:84: in function 'backward'
	train.lua:763: in function 'lossFun'
	train.lua:868: in function 'Train'
	train.lua:999: in main chunk
	[C]: in function 'dofile'
	...dri1/torch/install/lib/luarocks/rocks/trepl/scm-1/bin/th:150: in main chunk
	[C]: at 0x00406670

-----------------------------------------------------------------------------
**Part of my network** 
---------------------------------------------------------
Dropout= nn.Dropout(0.5,busy)
net= nn.Sequential()
-- conv layer-1
net:add(nn.SpatialConvolution(3, 64, 3, 3, 1, 1, 1, 1))
net:add(nn.ReLU())
**net:add(Dropout)**
net:add(nn.SpatialConvolution(64, 64, 3, 3, 1, 1, 1, 1))
net:add(nn.ReLU())
**net:add(Dropout)**
net:add(nn.SpatialAveragePooling(2, 2, 2, 2))-

-- conv layer-2
net:add(nn.SpatialConvolution(64, 128, 3, 3, 1, 1, 1, 1))
net:add(nn.ReLU())
**net:add(Dropout)**
net:add(nn.SpatialConvolution(128, 128, 3, 3, 1, 1, 1, 1))
net:add(nn.ReLU())
net:add(nn.SpatialAveragePooling(2, 2, 2, 2))

..................
---------------------------------------------------------------------
my batch size 10 and image dimension: 3x224x224
